File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.size()==0) return 0;
        int l=0,r=0,ans=0, n=s.size();
        map<char,int>st;
        while(r<n){
           if(st.find(s[r])!=st.end()){
             l= max(st[s[r]]+1,l);
           }
           st[s[r]]=r;
           ans= max(ans, r-l+1);
           r++;
        }
         return ans;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {
        var ans =0;
        var n = s.length;
        var l=0,r=0;
        var maxf=0;
        var count = new Map();
        while(r<n){
            count.set(s[r],(count.get(s[r])||0)+1);
            maxf= Math.max(maxf,count.get(s[r]));
            while (r - l + 1 - maxf > k) {
                count.set(s[l], count.get(s[l]) - 1);
                l++;
            }
            ans= Math.max(ans, r-l+1);
            r++;
        }
        return ans;
    }
}
